crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -346,24 +346,26 @@ void HMAC_verify_service_indicator(const EVP_MD *evp_md) {
   }
 }
 
-void PBKDF_verify_service_indicator(const EVP_MD *evp_md) {
-  // PBKDF with SHA1, SHA224, SHA256, SHA384, and SHA512 are approved.
+void PBKDF2_verify_service_indicator(const EVP_MD *evp_md, size_t password_len,
+                                    size_t salt_len, unsigned iterations) {
   //
-  // TODO (CryptoAlg-1270): FIPS 140 parameter requirements, per
-  // NIST SP800-132:
+  // FIPS 140 parameter requirements, per NIST SP800-132:
   //
-  // * key_len >= 14 bytes (112 bits)
+  // * password_len >= 14 bytes (112 bits)
   // * salt_len >= 16 bytes (128 bits), assuming its randomly generated
   // * iterations "as large as possible, as long as the time required to
   //   generate the key using the entered password is acceptable for the users."
-  //   (clearly we can't test for "as large as possible")
+  //   (clearly we can't test for "as large as possible"); NIST SP800-132
+  //   suggests >= 1000, but it's still not a requirement.
   switch (evp_md->type) {
     case NID_sha1:
```

Comment:
Is SHA1 still approved? SP 800-131A section 9 doesn't make it clear KDF is an approved use, does PBKDF2 need collision resistance?

Reply:
AFAIK SHA-1 is still in the approved list, presumably for "legacy" uses. I think the weaker collision resistance is mitigated somewhat by password/salt/iterations being in the mix.

Reply:
According to the NIST ACVP testing data, SHA-1 is approved: https://pages.nist.gov/ACVP/draft-celi-acvp-pbkdf.html#section-7.3.1